tests/integration/api.int.js

```diff
@@ -90,3 +114,143 @@ describe('API proxy endpoint integration tests', () => {
 		});
 	});
 });
+
+/*
+ * BEGIN 'POST' TESTS
+ */
+const mockPostPayload = {
+	queries,
+};
+
+const runMutationTest = ({ csrfHeaders, method, test, queries }) => server => {
+	csrfHeaders = csrfHeaders || getCsrfHeaders;
+	return csrfHeaders()
+		.then(([headerToken, cookieToken]) => {
+			const headers = {
+				'x-csrf-jwt': headerToken,
+				'content-type': 'application/x-www-form-urlencoded',
+				cookie: `x-csrf-jwt=${cookieToken}`,
+			};
+			const request = {
+				method,
+				url: `/mu_api${queries ? `?queries=${queries}` : ''}`,
+				payload: queries
+					? undefined
+					: querystring.stringify(mockPostPayload),
+				credentials: 'whatever',
+				headers,
+			};
+			return server.inject(request);
+		})
+		.then(test)
+		.then(() => server.stop())
+		.catch(err => {
+			server.stop();
+			throw err;
+		});
+};
+
+describe('API proxy POST endpoint integration tests', () => {
```

Comment:
Yeah, in this case I think that's alright because I don't _think_ we actually care about cases where the handlers may or may not be called - we only care that they _are_ called for the specific request methods that we've defined. For example, I don't think the tests should fail if the POST handler ended up delegating to the GET handler, so we don't need to test that the GET handler _is not called_ in the POST case. Essentially, I'd argue that the handlers are not designed to have an opinion on incorrect/irrelevant requests, so it's not necessary to verify the behavior of incorrect/irrelevant requests. Let me know if you have a suggestion on what to add, though.